### PR TITLE
Support GJF own import order

### LINKED ISSUE #1769: Support Google-Java-Formats own import order out of the box
As mentioned in #1680 spotless v6.18.0 unintentionally switched to gjf's own import order handling which was introduced as default also into the corresponding intellij plugin.
With spotless v6.19.0 the import order was changed back to the previous behavior of spotless prior v6.18.
I would like to see a simple config API change to the `googleJavaFormat()` definition which disables spotless own import order and uses the order from gjf.

### PR DESCRIPTION
**Pull request by Goooler**
Closes #1769.

### CODE REVIEW

File: lib/src/googleJavaFormat/java/com/diffplug/spotless/glue/java/GoogleJavaFormatFormatterFunc.java [hunk 1]
```diff
@@ -60,7 +65,7 @@ public String apply(@Nonnull String input) throws Exception {
 		// Issue #1679: we used to call ImportOrderer.reorderImports(String) here, but that is deprecated.
 		// Replacing the call with (the correct) reorderImports(String, Style) causes issues for Style.AOSP,
 		// so we force the style to GOOGLE for now (which is what the deprecated method did)
```
Comment (nedtwigg) [inline lib/src/googleJavaFormat/java/com/diffplug/spotless/glue/java/GoogleJavaFormatFormatterFunc.java:67]: Remove this comment.

**Review by nedtwigg — CHANGES_REQUESTED** [review]
Looks great! Just a few missing pieces

- root `CHANGES.md`
- update gjf docs in `plugin-gradle/README.md`
  - https://github.com/diffplug/spotless/blob/main/plugin-gradle/README.md#google-java-format

Not required, but it's such a small change, might be nice to add it into the maven plugin too

- https://github.com/diffplug/spotless/blob/main/plugin-maven/src/main/java/com/diffplug/spotless/maven/java/GoogleJavaFormat.java

File: lib/src/googleJavaFormat/java/com/diffplug/spotless/glue/java/GoogleJavaFormatFormatterFunc.java (hunk 1 above)
Comment (Goooler) [reply lib/src/googleJavaFormat/java/com/diffplug/spotless/glue/java/GoogleJavaFormatFormatterFunc.java]: 580d9c34708bdb863fffa049ccdfe1e7d4baadd4

**Review by nedtwigg — APPROVED** [review]
Thanks very much!

**Comment by nedtwigg** [comment]
Released in `plugin-gradle 6.21.0` and `plugin-maven 2.38.0`.

**Comment by bjrke** [comment]
I tried this out with the recently release version 6.21 of the gradle plugin.
It looks like the new setting `reorderImports(true)` doesn't have any effect on our setup. While others like aosp() or using an older version is working.

I get the following change in the import order when updating from 6.18.0 to 6.21.0:
```
          -
          -import·org.slf4j.Logger;
          -import·org.slf4j.LoggerFactory;
          -
           import·java.nio.file.Path;
           import·java.nio.file.Paths;
           import·java.util.Locale;
           import·java.util.Optional;
           import·java.util.stream.Stream;
          +import·org.slf4j.Logger;
          +import·org.slf4j.LoggerFactory;
```

File: plugin-gradle/src/main/java/com/diffplug/gradle/spotless/JavaExtension.java [hunk 2]
```diff
@@ -207,13 +208,19 @@ public GoogleJavaFormatConfig reflowLongStrings(boolean reflowLongStrings) {
 			return this;
 		}
 
+		public GoogleJavaFormatConfig reorderImports(boolean reorderImports) {
+			this.reorderImports = reorderImports;
```
Comment (bjrke) [inline plugin-gradle/src/main/java/com/diffplug/gradle/spotless/JavaExtension.java:212]: it think a 
```
replaceStep(createStep());
```
is missing here, see https://github.com/diffplug/spotless/issues/1871

Comment (Goooler) [reply plugin-gradle/src/main/java/com/diffplug/gradle/spotless/JavaExtension.java:212]: Fixed in #1872.

**Comment by Marcono1234** [comment]
If I understand the changes here correctly, the name `reorderImports` is a bit misleading because the imports are reordered anyway, regardless of the value. The parameter just affects which style is used for import ordering.

Would it make sense to rename the parameter (and deprecate the existing name) to make its behavior more clear? For example something like `useSameStyleForImportOrder` (or probably some other better name)

**Pull request merged**